src/main/java/core/basesyntax/service/impl/CsvDataReaderImpl.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.TransactionDto;
+import core.basesyntax.service.DataReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+public class CsvDataReaderImpl implements DataReader {
+    private CsvParser csvParser;
+
+    public CsvDataReaderImpl(CsvParser csvParser) {
+        this.csvParser = csvParser;
+    }
+
+    @Override
+    public List<TransactionDto> read(String filePath) {
+        try {
+            List<String> allLines = Files.readAllLines(Path.of(filePath));
+            allLines.remove(0);
+            return csvParser.parse(allLines);
```

Comment:
```suggestion
            return lines;
```
we can make the services more independent and call parser from main()

Reply:
refactored